genesyscloud/journey_views/resource_genesyscloud_journey_views_utils.go

```diff
@@ -0,0 +1,253 @@
+			followedBy := make([]platformclientv2.Journeyviewlink, len(followedBySlice))
+			for i, fb := range followedBySlice {
+				followedByMap, ok := fb.(map[string]interface{})
+				if !ok {
+					return nil, errors.New("followedBy element is not a map[string]interface{}")
+				}
+				followedBy[i] = buildJourneyViewLink(followedByMap)
+			}
+			element.FollowedBy = &followedBy
+		}
+		elements = append(elements, element)
+	}
+
+	return &elements, nil
+}
+
+func buildJourneyViewElementAttributes(attributesSlice []interface{}) platformclientv2.Journeyviewelementattributes {
+	var attributes platformclientv2.Journeyviewelementattributes
+	for _, elem := range attributesSlice {
+		if attributesMap, ok := elem.(map[string]interface{}); ok {
+			attributes.VarType = getStringPointerFromInterface(attributesMap["type"])
+			attributes.Id = getStringPointerFromInterface(attributesMap["id"])
+			attributes.Source = getStringPointerFromInterface(attributesMap["source"])
+		}
+	}
+	return attributes
+}
+
+func buildJourneyViewElementFilter(filterSlice []interface{}) platformclientv2.Journeyviewelementfilter {
+	var filter platformclientv2.Journeyviewelementfilter
+	for _, elem := range filterSlice {
+		if filterMap, ok := elem.(map[string]interface{}); ok {
+			filter.VarType = getStringPointerFromInterface(filterMap["type"])
+			if predicatesSlice, ok := filterMap["predicates"].([]interface{}); ok {
+				predicates := make([]platformclientv2.Journeyviewelementfilterpredicate, len(predicatesSlice))
+				for i, predicate := range predicatesSlice {
+					predicateMap, ok := predicate.(map[string]interface{})
+					if ok {
+						predicates[i] = buildJourneyviewelementfilterpredicate(predicateMap)
+					}
+				}
+				filter.Predicates = &predicates
+			}
+		}
+	}
+	return filter
+}
+
+func buildJourneyviewelementfilterpredicate(predicateMap map[string]interface{}) platformclientv2.Journeyviewelementfilterpredicate {
+	var predicate platformclientv2.Journeyviewelementfilterpredicate
+	predicate.Dimension = getStringPointerFromInterface(predicateMap["dimension"])
+	if valuesSlice, ok := predicateMap["values"].([]interface{}); ok {
+		values := make([]string, len(valuesSlice))
+		for i, value := range valuesSlice {
+			if stringValue, ok := value.(string); ok {
+				values[i] = stringValue
+			}
+		}
+		predicate.Values = &values
+	}
+	predicate.Operator = getStringPointerFromInterface(predicateMap["operator"])
+	predicate.NoValue = getBoolPointerFromInterface(predicateMap["noValue"])
+	return predicate
+}
+
+func buildJourneyViewLink(linkMap map[string]interface{}) platformclientv2.Journeyviewlink {
+	var link platformclientv2.Journeyviewlink
+	link.Id = getStringPointerFromInterface(linkMap["id"])
+	constraintWithinSlice, ok := linkMap["constraint_within"].([]interface{})
+	if ok {
+		link.ConstraintWithin = buildJourneyViewLinkTimeConstraint(constraintWithinSlice)
+	}
+	if !ok {
+		log.Printf("wrong type. correct type %T", constraintWithinSlice)
+	}
```

Comment:
This `!ok` block can be removed. How it is done for constraint_after below is perfect